src/containers/BooksForm.js

```diff
@@ -1,26 +1,60 @@
-import React from 'react';
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { createBook } from '../actions/index';
 
 const categories = ['Action', 'Biography', 'History', 'Horror', 'Kids', 'Learning', 'Sci-Fi'];
 
-const BooksForm = () => (
-  <form>
-    <div>
-      <label htmlFor="title">
-        ADD NEW BOOK
-        <input id="title" name="title" type="text" placeholder="Add title" />
-      </label>
-    </div>
-    <div>
-      <select id="categories">
-        <option>Category</option>
-        {
+const BooksForm = ({ addBook }) => {
+  const [book, setBook] = useState({ id: '', title: '', category: '' });
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    addBook(String(Math.floor(Math.random() * 1000000)), book.title, book.category);
+    setBook({ id: '', title: '', category: '' });
+  };
+
+  const handleChange = (event) => {
+    if (event.target.id === 'title') {
+      setBook({ ...book, title: event.target.value });
+    }
+    if (event.target.id === 'categories') {
+      setBook({ ...book, category: event.target.value });
+    }
+  };
+
+  return (
+    <form onSubmit={handleSubmit}>
+      <div>
+        <label htmlFor="title">
+          ADD NEW BOOK
+          <input id="title" name="title" type="text" placeholder="Add title" onChange={handleChange} value={book.title} />
+        </label>
+      </div>
+      <div>
+        <select id="categories" value={book.category} onChange={handleChange}>
+          <option value="" disabled>Category</option>
+          {
           categories
-            .map((category) => (<option key={category} value={category}>{category}</option>))
+            .map((category) => (
+              <option key={category} value={category}>
+                {category}
+              </option>
+            ))
         }
-      </select>
-    </div>
-    <button type="button">Add book</button>
-  </form>
-);
+        </select>
+      </div>
+      <button type="submit">Add book</button>
+    </form>
+  );
+};
```

Comment:
- [ ] Your application currently allows users to submit books with blank info i.e an object with keys that have no values for id, title, category. Kindly add some validation to not allow users to submit an entry if they have not provided all the required information.